sites/fast-website/src/app/components/side-navigation/side-navigation.ts

```diff
@@ -14,21 +19,30 @@ export class SideNavigation extends FASTElement {
     @observable
     public sectionArray: HTMLElement[] = [];
 
+    @observable
+    public objArray: sectionData[];
+
     public socialData: CommunityContentPlacementData[] = communityContentPlacementData.filter(
         x => x.header !== "Github"
     );
 
     loadScrollers() {
         this.sectionArray = Array.from(document.querySelectorAll("section[id]"));
 
+        this.objArray = this.sectionArray.map(x => {
+            return { section: x.id, value: 0 };
+        });
+
         let observer = new IntersectionObserver(
             entries => {
                 entries.forEach(entry => {
-                    const areaOnScreen =
-                        entry.intersectionRatio * entry.boundingClientRect.height;
-                    if (areaOnScreen > 0.5 * window.innerHeight) {
-                        this.currentSection = entry.target.id;
-                    }
+                    const intersectingSection: sectionData = this.objArray.find(
+                        ({ section }) => section === entry.target.id
+                    ) as sectionData;
+                    intersectingSection.value = entry.intersectionRatio;
+                    this.currentSection = this.objArray.reduce((prev, current) =>
```

Comment:
I think you can do this once after running through all the entries rather than after each entry update.

Reply:
I have moved the location of this so that it is run after all the entries instead of being run ever time.